### PR TITLE
Add a public field to TalkUrl

### PR DESCRIPTION
**Pull request by stefanor**
It's sometimes useful to have non-public URLs attached to a talk. E.g. the Video Conference URL that the speaker will use to present the talk.

This is often useful, but sometimes not *all* of the URLs, so let's make it easy to build templates that can display just the public URLs.

### CODE REVIEW

**Comment by hodgestar** [comment]
The intention of the docstring was to say that the fields would not be editable by the talk authors. They're there for the conference organizers to fill in. The intention wasn't to say they wouldn't be viewable (probably that could have been clearer).

**Comment by hodgestar** [comment]
What is the goal of the new `public` flag? If someone doesn't want the URL to be visible to the public, can it just be left out instead? And why would one want to hide a URL? Presumably a URL is already publicly accessible?

**Comment by stefanor** [comment]
> The intention of the docstring was to say that the fields would not be editable by the talk authors. They're there for the conference organizers to fill in. The intention wasn't to say they wouldn't be viewable (probably that could have been clearer).

Aha. Happy to restore it and tweak it for clarity, there.

> What is the goal of the new public flag?

Our use-case is URLs for Jitsi rooms, that we want to create for each talk and share with some people but not everyone. See   https://salsa.debian.org/debconf-team/public/websites/wafer-debconf/-/merge_requests/93
So, Metadata that we attach to talks that we don't show the whole world. Could use other things for that, too, like KV, but this did seem like the right mechanism. At least from my interpretation of the comment...

File: wafer/talks/models.py [hunk 1]
```diff
@@ -346,15 +346,13 @@ def can_review(self, user):
 
 @reversion.register()
 class TalkUrl(models.Model):
-    """An url to stuff relevant to the talk - videos, slides, etc.
-
-       Note that these are explicitly not intended to be exposed to the
-       user, but exist for use by the conference organisers."""
+    """An url to stuff relevant to the talk - videos, slides, etc."""
```
Comment (hodgestar) [inline wafer/talks/models.py:349]: Could we update this to:
```
""" A means for attaching relevant URLs to the talk (e.g. videos, slides).

    These URL associations are explicitly not intended to be edited by the author of the talk, but
    rather by the conference organizers.

    URLs with public set to False are not publicly visible.
"""
```

File: wafer/talks/models.py [hunk 2]
```diff
@@ -346,15 +346,13 @@ def can_review(self, user):
...
 
     description = models.CharField(max_length=256)
     url = models.URLField()
     talk = models.ForeignKey(Talk, related_name='urls',
                              on_delete=models.CASCADE)
+    public = models.BooleanField(default=False)
```
Comment (hodgestar) [inline wafer/talks/models.py:355]: Given that the previous behaviour was equivalent to `public=True` should the default not be True?

Comment (stefanor) [reply wafer/talks/models.py:355]: Sounds good to me

**Review by hodgestar — APPROVED** [review]

**Pull request merged**